source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4098,10 +4100,29 @@ export class ToggleVisibilityAction extends Action {
     // change introduced in TS 3.1 wrt d.ts generation. DO NOT CHANGE
     static readonly JsonTypeName: "Action.ToggleVisibility" = "Action.ToggleVisibility";
 
+    private updateAriaControlsAttribute() {
+        // apply aria labels to make it clear which elements this action will toggle
+        if (this.targetElements) {
+            const elementIds = Object.keys(this.targetElements);
+
+            if (elementIds.length > 0) {
+                this._renderedElement?.setAttribute("aria-controls", elementIds.join(" "));
```

Comment:
I have made a point not to use the ? operator to ensure consuming applications will not fail at compile time if they are not using a recent enough version of Typescript. I think we should continue not using it for now, but if we are going to use it we need to at least check with Teams that it's OK with them.

Reply:
I don't think this works. `elementIds` holds the Ids of the target AC elements, not the Ids of DOM element. For this to work, we need to dynamically set the Ids of the target DOM elements to be the same as the Ids of the target AC element or simply always assign the AC Id to all rendered DOM elements in `CardElement.render()`. Given the latter has been an ask from several developers, I suggest we do that and document it.